File: easy/backspace_compare_test.go
package easy

import (
	"testing"
)

func TestBackspaceCompare(t *testing.T) {
	cases := []struct {
		s        string
		t        string
		expected bool
	}{
		{"y#fo##f", "y#f#o##f", true},
		{"ab#c", "ad#c", true},
		{"ab##", "c#d#", true},
		{"a#c", "b", false},
		{"a", "a", true},
		{"a", "b", false},
		{"#", "#", true},
		{"#a", "a", true},
		{"a#", "", true},
		{"a#", "#", true},
		{"##", "#", true},
		{"###", "##", true},
		{"a##", "#", true},
		{"a##", "", true},
		{"a#b", "b", true},
		{"a#b", "a", false},
		{"abc", "abc", true},
		{"abc", "abd", false},
		{"ab#c", "ab#c", true},
		{"ab##c", "c", true},
		{"a#b#c#", "", true},
		{"a#b#c#d", "d", true},
		{"a#b#c#d", "e", false},
		{"#a#b#", "", true},
		{"bxj##tw", "bxj##tw", true},
		{"bxj##tw", "bxk##tw", true},
		{"xywrrmp", "xywrrmu", false},
		{"xywrrmp", "xywrrmp", true},
		{"a###b", "b", true},
		{"a###b", "a", false},
		{"####", "", true},
		{"####", "####", true},
		{"a####", "#", true},
		{"ab####", "##", true},
		{"nzp#o#g", "b#nzp#o#g", true},
		{"bbbextm", "bbb#extm", false},
		{"ab#c#", "a", true},
		{"a#a#a#", "", true},
		{"abc##", "a", true},
		{"abc###", "", true},
		{"", "#", true},
		{"#", "", true},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := BackspaceCompare(c.s, c.t)
			if result != c.expected {
				t.Errorf("BackspaceCompare(%q, %q) == %t, expected %t", c.s, c.t, result, c.expected)
			}

		})

	}
}

File: src/test/java/easy/BackspaceCompareTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class BackspaceCompareTest {
  BackspaceCompare instance = new BackspaceCompare();

  @ParameterizedTest
  @MethodSource("dataBackspaceCompare")
  public void testBackspaceCompare(String s, String t, boolean expected) {

    boolean result = instance.solve(s, t);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataBackspaceCompare() {
    return Stream.of(
        Arguments.of("y#fo##f", "y#f#o##f", true),
        Arguments.of("ab#c", "ad#c", true),
        Arguments.of("ab##", "c#d#", true),
        Arguments.of("a#c", "b", false),
        Arguments.of("a", "a", true),
        Arguments.of("a", "b", false),
        Arguments.of("#", "#", true),
        Arguments.of("#a", "a", true),
        Arguments.of("a#", "", true),
        Arguments.of("a#", "#", true),
        Arguments.of("##", "#", true),
        Arguments.of("###", "##", true),
        Arguments.of("a##", "#", true),
        Arguments.of("a##", "", true),
        Arguments.of("a#b", "b", true),
        Arguments.of("a#b", "a", false),
        Arguments.of("abc", "abc", true),
        Arguments.of("abc", "abd", false),
        Arguments.of("ab#c", "ab#c", true),
        Arguments.of("ab##c", "c", true),
        Arguments.of("a#b#c#", "", true),
        Arguments.of("a#b#c#d", "d", true),
        Arguments.of("a#b#c#d", "e", false),
        Arguments.of("#a#b#", "", true),
        Arguments.of("bxj##tw", "bxj##tw", true),
        Arguments.of("bxj##tw", "bxk##tw", true),
        Arguments.of("xywrrmp", "xywrrmu", false),
        Arguments.of("xywrrmp", "xywrrmp", true),
        Arguments.of("a###b", "b", true),
        Arguments.of("a###b", "a", false),
        Arguments.of("####", "", true),
        Arguments.of("####", "####", true),
        Arguments.of("a####", "#", true),
        Arguments.of("ab####", "##", true),
        Arguments.of("nzp#o#g", "b#nzp#o#g", true),
        Arguments.of("bbbextm", "bbb#extm", false),
        Arguments.of("ab#c#", "a", true),
        Arguments.of("a#a#a#", "", true),
        Arguments.of("abc##", "a", true),
        Arguments.of("abc###", "", true),
        Arguments.of("", "#", true),
        Arguments.of("#", "", true));
  }
}
